### PR TITLE
Add Tree Postorder Traversal in TypeScript

### LINKED ISSUE #2551: Tree Postorder Traversal
- [x] C
- [x] C++
- [ ] C# - @sharmakeshav10
- [ ] CoffeeScript
- [ ] Dart - @Amitsharma45
- [x] Go
- [x] Java
- [x] JS
- [ ] Kotlin - @PrachieNaik
- [x] Php
- [x] Python
- [ ] Readme - @SimranMakhija7
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @shivamgupta57121

### PR DESCRIPTION
**Pull request by shivamgupta57121**
Fixes #2551

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Add code for Tree Postorder Traversal in TypeScript 

#### Languages Used:
- TypeScript

#### Files Added:
- Tree_Postorder_Traversal.ts

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by shivamgupta57121** [comment]
Please review my PR @jainaman224 @singh-shreya6

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// Recursive Function to call itself on left child, then on right child and then print value of current node 
+function postOrderTraversal(currentNode: treeNodes) 
+{
+    if (!currentNode)
+        return;
+    postOrderTraversal(currentNode.left); 
+    postOrderTraversal(currentNode.right);
+    console.log(currentNode.val + " -> ");
+}
+
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.ts:55]: remove the extra line

Comment (shivamgupta57121) [reply Tree_Postorder_Traversal/Tree_Postorder_Traversal.ts:54]: Removed

**Comment by shivamgupta57121** [comment]
@asha15 Removed the extra line. Please review it.

**Comment by raghav-dalmia** [comment]
Squash your commits.

**Comment by shivamgupta57121** [comment]
I have squashed the changes. @raghav-dalmia

**Comment by shivamgupta57121** [comment]
I have squashed the changes. Please review it and merge @raghav-dalmia @asha15

**Pull request merged**